src/peft/tuners/lora/layer.py

```diff
@@ -867,6 +870,9 @@ def __init__(
         LoraLayer.__init__(self, base_layer)
 
         self._active_adapter = adapter_name
+        self._kernel_dim = base_layer.weight.dim()
+        self._convd = self._kernel_dim - 2
```

Comment:
Do we really need a separate attribute for this or can we not just use `_kernel_dim`?

Reply:
Certainly using _kernel_dim would work; I thought this would make it more obvious what was going on. I just updated the branch.